junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/AbstractOrderingVisitor.java

```diff
@@ -178,12 +188,18 @@ private Map<Object, Integer> distinctWrappersToIndex(List<?> wrappers) {
 			return toIndex;
 		}
 
-		private void logDescriptorsAddedWarning(int number) {
-			logger.warn(() -> this.descriptorsAddedMessageGenerator.generateMessage(number));
+		private void logDescriptorsAddedWarning(TestDescriptor parentTestDescriptor, int number) {
+			reportWarning(parentTestDescriptor, this.descriptorsAddedMessageGenerator.generateMessage(number));
+		}
+
+		private void logDescriptorsRemovedWarning(TestDescriptor parentTestDescriptor, int number) {
+			reportWarning(parentTestDescriptor, this.descriptorsRemovedMessageGenerator.generateMessage(number));
 		}
 
-		private void logDescriptorsRemovedWarning(int number) {
-			logger.warn(() -> this.descriptorsRemovedMessageGenerator.generateMessage(Math.abs(number)));
+		private void reportWarning(TestDescriptor parentTestDescriptor, String message) {
+			AbstractOrderingVisitor.this.issueReporter //
+					.reportIssue(EngineDiscoveryIssue.builder(Severity.WARNING, message) //
+							.source(parentTestDescriptor.getSource()));
```

Comment:
:speech_balloon: Another example of a custom "post-discovery" validation that reports a warning for issues during class/method ordering.